docs2/site/docs/getting-started/custom-scalars.md

```diff
@@ -183,105 +213,303 @@ public class ExampleSchema : Schema
             throw new FormatException($"Failed to parse {nameof(Vector3)} from input '{vector3InputString}'. Input should be a string of three comma-separated floats in X Y Z order, ex. 1.0,2.0,3.0");
         }
     }
+
+    return ThrowValueConversionError(value);
 }
 ```
 
-6. Prepare to support conversion of `Vector3` to an AST node. This allows GraphQL.NET to treat values parsed from variables the same as arguments, which arrive for execution as AST nodes. 
+6. Implement `ScalarGraphType.Serialize` so `Vector3` instances can be sent to the client.
+Keep in mind that a `null` value also required to be handled.
 
-Implement `ValueNode<Vector3>`. Instances of `Vector3` parsed from variables will be wrapped in this type during execution.
+It is recommended that the type of data this method returns match the same type as
+is handled within `ParseValue`. In this case, serialization of `Vector3` is serialized as
+a string, which matches the expected data type within `ParseValue`.
 
 ```csharp
-public class Vector3Value: ValueNode<Vector3>
+// In Vector3Type
+
+public override object Serialize(object value)
 {
-    public Vector3Value(Vector3 value)
+    if (value == null)
+        return null;
+
+    if (value is Vector3 vector3)
     {
-        Value = value;
+        return $"{vector3.X},{vector3.Y},{vector3.Z}";
     }
 
-    protected override bool Equals(ValueNode<Vector3> node)
-    {
-        return Value.Equals(node.Value);
+    return ThrowSerializationError(value);
+}
+```
+
+7. Override `ScalarGraphType.ToAST` if necessary.
+
+The infrastructure converts default field values to AST representations during initialization
+in order to verify that the default values are valid within an AST tree. The default implementation
+calls `Serialize` to convert the value to its client-side equivalent, then embeds it into an
+appropriate AST node based on the data type returned from `Serialize`. It is likely you will
+only need to override this method if you are creating a custom scalar that returns enumeration
+values, or if you are returning structured data.
+
+In this example, you created a custom scalar. In summary:
+
+- Create a class for the server-side representation of the scalar
```

Comment:
What class ? ScalarGraphType or data-class ?

Reply:
Yeah I didn't write this...

Reply:
```suggestion
- Create a data class for the server-side representation of the scalar
```